code/game/objects/items/cigs_lighters.dm

```diff
@@ -1025,28 +1025,28 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		reagents.flags |= NO_REACT
 		STOP_PROCESSING(SSobj, src)
 
-/obj/item/clothing/mask/vape/proc/handle_reagents()
+/obj/item/clothing/mask/vape/proc/handle_reagents(delta_time)
 	if(!reagents.total_volume)
 		return
 
 	var/mob/living/carbon/vaper = loc
 	if(!iscarbon(vaper) || src != vaper.wear_mask)
-		reagents.remove_any(REAGENTS_METABOLISM)
+		reagents.remove_any(REAGENTS_METABOLISM * delta_time * 0.5)
 		return
 
 	if(reagents.get_reagent_amount(/datum/reagent/fuel))
 		//HOT STUFF
-		vaper.adjust_fire_stacks(2)
+		vaper.adjust_fire_stacks(2 * delta_time * 0.5)
 		vaper.IgniteMob()
 
 	if(reagents.get_reagent_amount(/datum/reagent/toxin/plasma)) // the plasma explodes when exposed to fire
 		var/datum/effect_system/reagents_explosion/e = new()
-		e.set_up(round(reagents.get_reagent_amount(/datum/reagent/toxin/plasma) / 2.5, 1), get_turf(src), 0, 0)
+		e.set_up(round(reagents.get_reagent_amount(/datum/reagent/toxin/plasma) / 2.5, 1), get_turf(src), 0, 0) //this is an explosion that destroys the vape, so it should NOT be affected by delta_time
 		e.start()
 		qdel(src)
 
-	if(!reagents.trans_to(vaper, REAGENTS_METABOLISM, methods = INGEST, ignore_stomach = TRUE))
-		reagents.remove_any(REAGENTS_METABOLISM)
+	if(!reagents.trans_to(vaper, REAGENTS_METABOLISM * delta_time * 0.5, methods = INGEST, ignore_stomach = TRUE))
+		reagents.remove_any(REAGENTS_METABOLISM * delta_time * 0.5)
```

Comment:
just make these variables or something

Reply:
eh, REAGENTS_METABOLISM * delta_time * 0.5 is basically just used here, and no vape subtypes exist yet

Reply:
scalability exists for the future, not just for current convenience.